src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -355,7 +356,11 @@ public void onFailure(Exception e) {
                 listener.onFailure(new ElasticsearchSecurityException(err, RestStatus.FORBIDDEN));
                 return;
             }
-        } catch (Throwable e) {
+        }
+        catch (InvalidIndexNameException err) {
```

Comment:
should it catch `ElasticsearchException` instead?

Reply:
Done.